include/range/v3/view/iota.hpp

```diff
@@ -163,6 +163,25 @@ namespace ranges
                     :  static_cast<D>(v0 - v1);
             }
 
+            template<typename Val, CONCEPT_REQUIRES_(SignedIntegral<Val>())>
+            RANGES_CXX14_CONSTEXPR
+            iota_difference_t<Val> ints_distance_(Val from, Val to) noexcept {
+                using D = iota_difference_t<Val>;
+                RANGES_EXPECT(from <= to);
+                RANGES_EXPECT((static_cast<D>(to) - static_cast<D>(from))
+                              < std::numeric_limits<D>::max());
+                return static_cast<D>(to) - static_cast<D>(from);
+            }
+
+            template<typename Val, CONCEPT_REQUIRES_(UnsignedIntegral<Val>())>
+            RANGES_CXX14_CONSTEXPR
+            iota_difference_t<Val> ints_distance_(Val from, Val to) noexcept {
+                using D = iota_difference_t<Val>;
+                RANGES_EXPECT(static_cast<D>(to - from) < std::numeric_limits<D>::max());
```

Comment:
See my comment above. While this check is probably wrong, `to - from < std::numeric_limits<range_size_t>::max()` or otherwise computing the size will overflow and give `0`, so I think fixing the check here makes sense. Otherwise we end up with ranges of size `0` that are iterable ...